merge_fastqs.py

```diff
@@ -1,85 +1,43 @@
-# Modified version of https://github.com/SciLifeLab/standalone_scripts/blob/master/merge_and_rename_NGI_fastq_files.py
-# to fit the NGI Uppsala standard
-
+import argparse
 import re
 import os
-import sys
 import shutil
-import argparse
-import collections
 
-def merge_files(input_dir, dest_dir):
+arg_parser = argparse.ArgumentParser(description=""" Merges all fastq-files that match the standard name pattern per sample and read. Looks through the given dir and subdirs.""")
+arg_parser.add_argument("--input_dir", metavar='Input directory', required=True, help="Base directory for the fastq files that should be merged. ")
+arg_parser.add_argument("--output_dir", metavar='Output directory', required=True, help="Path for output of merged files.")
+args = arg_parser.parse_args()
 
-    #Gather all fastq files in inputdir and its subdirs
-    fastq_files=[]
-    for subdir, dirs, files in os.walk(input_dir):
-        for fastq in  files:
-            if fastq.endswith('.fastq.gz'):
-                fastq_files.append(os.path.join(subdir, fastq))
+sample_pattern = re.compile(r"^(.+)_S[0-9]+_L00[1-8]_(R[1-2])_.+\.fastq\.gz$")
```

Comment:
This is a bit picky, but how about:

```suggestion
sample_pattern = re.compile(r"^(.+)_S[0-9]+_L[0-9]{3}_(R[0-9]+)_.+\.fastq\.gz$")
```

Reply:
Yeah, that pattern is more general 👍